onnx/test/test_external_data.py

```diff
@@ -285,9 +285,12 @@ def create_test_model_both(self):  # type: () -> Tuple[Text, ModelProto]
     # 2GB models would fail onnx.checker with ModelProto but pass with model path
     def test_check_model_by_model(self):  # type: () -> None
         model = onnx.load_model(self.model_filename, load_external_data=False)
-        with pytest.raises(ValueError):
+        try:
             load_external_data_for_model(model, self.temp_dir)  # Exceeds maximum protobuf
-            checker.check_model(model)  # checker catches 2GB models as well
+        except MemoryError:
+            print("Warning: Because of the hardware limitation (memory), this test was not executed.")
+            return
```

Comment:
take a look at windows CI - it is printing this warning but still marks the pytest as failed

Reply:
Thank you for catching this. Just modified.

Reply:
In the latest windows CI run there is no warning message saying this test was skipped. Can you fix the CI.
It is OK to skip this test on Windows because of hardware limitation but we need to explicitly call it out.

Reply:
By default `pytest` prints stdout only if the test fails. Even though adding `-s` to `pytest` can capture the stdout from tests, it will apply to all tests. Is it acceptable?

Reply:
instead of catching mem error can you simply skip this test on windows something like this : https://docs.pytest.org/en/stable/skipping.html#skipping-test-functions and see if pytest can add skip tests in summary

Reply:
@askhade do you mean something like this?
```
@pytest.mark.skipif(os.name == 'nt', reason="Because of the hardware limitation (memory) in Windows-CI, this test was not executed.")
```
Done, but the linux failed. It seems like the old problem is happening again. https://github.com/onnx/onnx/pull/2933

Reply:
did you try this :  @pytest.mark.skipif(sys.platform.startswith("win"), reason="Because of the hardware limitation (memory) in Windows-CI, this test was not executed.")

Reply:
Linux failure is because of a type check error : 
onnx/test/test_external_data.py:288: error: Untyped decorator makes function "test_check_model_by_model" untyped

Reply:
Thanks for the catch. Just fixed. That test has been successfully skipped on Windows.